test/extensions/filters/http/wasm/test_data/shared_queue_rust.rs

```diff
@@ -38,24 +37,50 @@ impl RootContext for TestRoot {
             }
         }
     }
+
+    fn get_type(&self) -> Option<ContextType> {
+        Some(ContextType::HttpContext)
+    }
+
+    fn create_http_context(&self, _: u32) -> Option<Box<dyn HttpContext>> {
+        Some(Box::new(TestStream {
+            queue_id: self.queue_id,
+        }))
+    }
 }
 
-struct TestStream;
+struct TestStream {
+    queue_id: Option<u32>,
+}
 
 impl Context for TestStream {}
 
 impl HttpContext for TestStream {
     fn on_http_request_headers(&mut self, _: usize) -> Action {
+        if self.resolve_shared_queue("", "bad_shared_queue").is_none() {
+            warn!("onRequestHeaders not found self/bad_shared_queue");
+        }
         if self
             .resolve_shared_queue("vm_id", "bad_shared_queue")
             .is_none()
         {
-            warn!("onRequestHeaders not found bad_shared_queue");
+            warn!("onRequestHeaders not found vm_id/bad_shared_queue");
+        }
+        if self
+            .resolve_shared_queue("bad_vm_id", "bad_shared_queue")
+            .is_none()
+        {
+            warn!("onRequestHeaders not found bad_vm_id/bad_shared_queue");
+        }
+        if Some(self.resolve_shared_queue("", "my_shared_queue")) == Some(self.queue_id) {
+            warn!("onRequestHeaders found self/my_shared_queue");
         }
-        if let Some(queue_id) = self.resolve_shared_queue("vm_id", "my_shared_queue") {
-            self.enqueue_shared_queue(queue_id, Some(b"data1")).unwrap();
-            warn!("onRequestHeaders enqueue Ok");
+        if Some(self.resolve_shared_queue("vm_id", "my_shared_queue")) == Some(self.queue_id) {
+            warn!("onRequestHeaders found vm_id/my_shared_queue");
         }
+        self.enqueue_shared_queue(self.queue_id.unwrap(), Some(b"data1"))
```

Comment:
nit: where does assignment to queue_id happens? I see that we satisfy `self.resolve_shared_queue("vm_id", "my_shared_queue")) == Some(self.queue_id)` in the happy case so this isn't a problem at all, but it looks different to me compared to the cpp case where assignment to `token` explicitly happens.

Reply:
Rust and C++ tests often aren't direct translations.

`queue_id` is assigned in the RootContext: 
https://github.com/envoyproxy/envoy/blob/c4861ffad700a7281c4fce4d886c82ccef572dbb/test/extensions/filters/http/wasm/test_data/shared_queue_rust.rs#L20-L23
and inherited in the HttpContext's constructor:
https://github.com/envoyproxy/envoy/blob/c4861ffad700a7281c4fce4d886c82ccef572dbb/test/extensions/filters/http/wasm/test_data/shared_queue_rust.rs#L45-L49

Notably, this is the "correct" pattern for using shared queue within the plugin, and `proxy_resolve_shared_queue` was added to support cross-VM queue resolution, but it seems that people are misusing it for in-plugin resolution, and this PR is helping with that.

Reply:
I missed the `vm_start` part, thanks!